lib/dockerregistry/inventory.go

```diff
@@ -1531,6 +1572,152 @@ func (sc *SyncContext) GarbageCollect(
 	}
 }
 
+func supportedMediaType(v string) (cr.MediaType, error) {
+	switch cr.MediaType(v) {
+	case cr.DockerManifestList:
+		return cr.DockerManifestList, nil
+	case cr.DockerManifestSchema1:
+		return cr.DockerManifestSchema1, nil
+	case cr.DockerManifestSchema1Signed:
+		return cr.DockerManifestSchema1Signed, nil
+	case cr.DockerManifestSchema2:
+		return cr.DockerManifestSchema2, nil
+	default:
+		return cr.MediaType(""), fmt.Errorf("unsupported MediaType %s", v)
+	}
+}
+
+// ClearRepository wipes out all Docker images from a registry! Use with caution.
+// nolint[gocyclo]
+//
+// TODO: Maybe split this into 2 parts, so that each part can be unit-tested
+// separately (deletion of manifest lists vs deletion of other media types).
+func (sc *SyncContext) ClearRepository(
+	regName RegistryName,
+	mfest Manifest,
+	mkProducer func(RegistryContext, ImageName, Digest) stream.Producer,
+	customProcessRequest *ProcessRequest) {
+
+	// deleteRequstsPopulator returns a PopulateRequests that
+	// varies by a predicate. Closure city!
+	var deleteRequstsPopulator func(func(cr.MediaType) bool) PopulateRequests = func(predicate func(cr.MediaType) bool) PopulateRequests {
```

Comment:
nit. Let's pull out these anonymous functions here and throughout this PR into standalone private functions. Its hard to follow a higher level flow of what the code is generally doing.

Reply:
ACK. Unfortunately I am not sure how to do this gracefully. The main reason we are passing functions around like this (e.g. why we have the PopulateRequests type to begin with) is because for unit tests we do not want to actually go over the real network for some of these functions.

For the case here, we use deleteRequestsPopulator to create network requests for only those manifests that are/are-not Manifest Lists.